### DISCUSSION TITLE
Loading a Relation to a Many-to-Many Relation in PostgreSQL via Prisma

### DISCUSSION THREAD

**Discussion by n-glaz** [prisma/prisma#14748] [Q&A] [answered]
I am new to Prisma and have a slightly sophisticated `many-to-many` relation in `PostgreSQL` that I wanna use in my Prisma Schema, but before executing I wanna make sure I can load this stuff up easily.

Here is the overview:

1. `Users` have a **Many-to-Many** relationship with `Missions`, through a `UsersOnMissions` model.
2. `Missions` **Have Many** `MissionTasks`
3. Individual `Users` are assigned `MissionTasks` via their UsersOnMissions `id`
4. `Users` **Have Many** `Mission Tasks` through `UsersOnMissions`

I am curious if the relation detailed in **4** above will be easy to load via Prisma, e.g. when I load a `User` I want to be able to load all of their `MissionTasks` easily on one view

**Comment by austincrim** [comment, marked as answer]
Hey @n-glaz 👋,

I don't foresee any issues doing that specific query. The best way to see the query ergonomics yourself would be to write out the Prisma schema and try it 🙂.

**Reply by n-glaz** [reply]
Thanks a bunch Austin! I've got the Schema ready to go, just wanted to get some confirmation to justify my many-to-many table implementation :)

**Reply by n-glaz** [reply]
@austincrim So regarding this related `through` relationship, is there a way to declare that nested `@relation` in `prisma.schema`, some syntax for like @relation(through [ManyToManyModelName]) e.g.:

```typescript
model User {
   // other fields omitted
  missions                             UsersOnMissions[]
}
model Mission {
  // other fields omitted
  users                             UsersOnMissions[]
}

model UsersOnMissions {
  id             Int           @id @default(autoincrement())
  user        User          @relation(fields: [userId], references: [id])
  userId      Int 
  mission       Mission           @relation(fields: [missionId], references: [id])
  missionId       Int
}

model MissionTask {
  //
  createdByUser      User    @relation(through MissionUser?)
  assignedToUser    User   @relation(through  MissionUser?)
}
```

**Reply by austincrim** [reply]
We do not support anything like that right now, but we have an https://github.com/prisma/prisma/issues/2186 for it.

Feel free to add your 👍 or specific use case in that thread.

**Reply by n-glaz** [reply]
got it, thanks again Austin!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
